### PR TITLE
MOVING TO TYPESCRIPT

### PR DESCRIPTION
**Pull request by tanelso2**

### CODE REVIEW

File: formatting/discordFormat.ts [hunk 1]
```diff
@@ -50,7 +54,7 @@ class DiscordFormat {
      * @param {Game} - The game to create an embed field for.
      * @return {Object} - The embed field to add for this game.
      */
-    makeMatchupField(game) {
+    makeMatchupField(game: Game): any {
```
Comment (rcrobert) [inline formatting/discordFormat.ts:57]: `string`?

File: generator/directory.ts [hunk 2]
```diff
@@ -90,9 +90,13 @@ const colors = [
     "light ${colour}"
 ];
 
-const numbersFromOneToFive = [...Array(5).keys()].map(x => (x + 1).toString());
+const numbersFromOneToFive = ["1", "2", "3", "4", "5"];
```
Comment (rcrobert) [inline generator/directory.ts:93]: But but but what if we wanted `numbersFromOneToFive` to actually include numbers 1..N this is totally inflexible! /s

File: index.ts [hunk 3]
```diff
@@ -242,7 +257,7 @@ client.on('message', message => {
     };
 
 
-    if (message.mentions.has(client.user, mentionsOptions)) {
+    if (message.mentions.has(client.user!, mentionsOptions)) {
```
Comment (rcrobert) [inline index.ts:260]: What's `!` mean here? This is new to me.

**Review by rcrobert — APPROVED** [review]
I like it

File: index.ts (hunk 3 above)
Comment (tanelso2) [reply index.ts:260]: Fuck you compiler, I know this property is type `Nullable<T>` but I just want a fucking T

File: formatting/discordFormat.ts (hunk 1 above)
Comment (tanelso2) [reply formatting/discordFormat.ts:57]: It's not though... I should define an interface but I got lazy

File: generator/directory.ts (hunk 2 above)
Comment (tanelso2) [reply generator/directory.ts:93]: Oh yeah this didn't play well with typescript for some reason? Idk what the problem was

**Pull request merged**